src/interfaces/simulation.ts

```diff
@@ -38,84 +44,140 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     options: SimulationOptions = {}
   ): Promise<TransactionOutcome> {
     const signer = this.ctx.provider.write.getSigner(from);
-    const zapProtocol = PickleJars.includes(toVault) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-    const vaultContract =
-      zapProtocol === ZapProtocol.PICKLE
-        ? new PickleJarContract(toVault, signer)
-        : new YearnVaultContract(toVault, signer);
-
-    const underlyingToken = await vaultContract.token().catch(() => {
-      throw new EthersError("failed to fetch token", EthersError.FAIL_TOKEN_FETCH);
-    });
-    const isZapping = underlyingToken !== sellToken;
-    let forkId: string | undefined;
-    let simulateDeposit: (save: boolean) => Promise<TransactionOutcome>;
 
-    if (isZapping) {
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
+    const vaultContract = this.getVaultContract({ toVault, signer });
 
-      let needsApproving: boolean;
+    const { isZappable, zappableType } = checkZappability({ chainId: this.chainId, token: options.token });
```

Comment:
Here we are checking if the token is zappable, but we should also be validating the vault accepts the same zap contract of the token.

Reply:
Done with https://github.com/yearn/yearn-sdk/pull/280/commits/7cec9684ce1efad411423b46474d0eb16f3963df